File: app/src/main/java/com/yao/zhihudaily/util/AssetsUtil.java
package com.yao.zhihudaily.util;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

/**
 * @author Yao
 * @date 2020/02/19
 */
public class AssetsUtil {

    public static String getString(String fileName) {
        try {
            InputStream is = ResUtil.getResources().getAssets().open(fileName);

            int length = is.available();
            byte[] buffer = new byte[length];
            is.read(buffer);
            return new String(buffer, StandardCharsets.UTF_8);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

}

File: app/src/main/java/com/yao/zhihudaily/util/AssetsUtil.kt
package com.yao.zhihudaily.util

import com.yao.zhihudaily.util.ResUtil.getResources
import java.io.IOException
import java.nio.charset.StandardCharsets

class AssetsUtil {

    fun getString(fileName: String): String? {
        try {
            val inputStream = getResources().assets.open(fileName)
            val length = inputStream.available()
            val buffer = ByteArray(length)
            inputStream.read(buffer)
            return String(buffer, StandardCharsets.UTF_8)
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return null
    }
}